modules/@apostrophecms/doc-type/index.js

```diff
@@ -506,7 +506,27 @@ module.exports = {
       //
       // `query.field` will contain the schema field definition for
       // the relationship the user is attempting to match titles from.
+      getRelationshipQueryBuilderChoicesProjection(query) {
+        return {
+          title: 1,
```

Comment:
* Leaving out type is never a good idea.
* Include `_url` just for good measure. It might not seem obviously relevant, but its implementation substitutes all the fields necessary to really know what document you're dealing with, without bringing in anything heavy.
* I just recently added support for displaying thumbnails when image pieces are the choices. Please make sure that still works. Possibly by extending this method for the image piece type so that it includes the attachment. If I'm right that it is using this new code now.